src/ToricVarieties/ToricDivisorClasses/constructors.jl

```diff
@@ -0,0 +1,208 @@
+
+Construct the toric divisor class associated to a list of integers which specify an element of the class group of the normal toric variety `v`.
+
+# Examples
+```jldoctest
+julia> P2 = projective_space(NormalToricVariety, 2)
+A normal, non-affine, smooth, projective, gorenstein, fano, 2-dimensional toric variety without torusfactor
+
+julia> tdc = ToricDivisorClass(P2, class_group(P2)([fmpz(1)]))
+A divisor class on a normal toric variety
+```
+"""
+ToricDivisorClass(v::AbstractNormalToricVariety, coeffs::Vector{fmpz}) = ToricDivisorClass(v, class_group(v)(coeffs))
+
+@doc Markdown.doc"""
+    ToricDivisorClass(v::AbstractNormalToricVariety, coeffs::Vector{Int})
+
+Construct the toric divisor class associated to a list of integers which specify an element of the class group of the normal toric variety `v`.
+
+# Examples
+```jldoctest
+julia> P2 = projective_space(NormalToricVariety, 2)
+A normal, non-affine, smooth, projective, gorenstein, fano, 2-dimensional toric variety without torusfactor
+
+julia> tdc = ToricDivisorClass(P2, class_group(P2)([1]))
+A divisor class on a normal toric variety
+```
+"""
+ToricDivisorClass(v::AbstractNormalToricVariety, coeffs::Vector{Int}) = ToricDivisorClass(v, class_group(v)([fmpz(c) for c in coeffs]))
+
+
+######################
+# 3: Special constructors
+######################
+
+@doc Markdown.doc"""
+    ToricDivisorClass(td::ToricDivisor)
+
+Construct the toric divisor class associated to the element ... of the class group of the normal toric variety `v`.
+
+# Examples
+```jldoctest
+julia> P2 = projective_space(NormalToricVariety, 2)
+A normal, non-affine, smooth, projective, gorenstein, fano, 2-dimensional toric variety without torusfactor
+
+julia> td = ToricDivisor(P2, [1,2,3])
+A torus-invariant, non-prime divisor on a normal toric variety
+
+julia> tdc = ToricDivisorClass(td)
+A divisor class on a normal toric variety
+```
+"""
+function ToricDivisorClass(td::ToricDivisor)
+    f = map_from_torusinvariant_weil_divisor_group_to_class_group(toric_variety(td))
+    class = f(sum(coefficients(td) .* gens(domain(f))))
+    return ToricDivisorClass(toric_variety(td), class)
+end
+
+
+########################
+# 4: Addition and scalar multiplication
+########################
+
+@doc Markdown.doc"""
+    Base.:+(tdc1::ToricDivisorClass, tdc2::ToricDivisorClass)
+
+Return the sum of the toric divisor classes `tdc1` and `tdc2`.
+
+# Examples
+```jldoctest
+julia> P2 = projective_space(NormalToricVariety, 2)
+A normal, non-affine, smooth, projective, gorenstein, fano, 2-dimensional toric variety without torusfactor
+
+julia> tdc1 = ToricDivisorClass(P2, [1])
+A divisor class on a normal toric variety
+
+julia> tdc2 = ToricDivisorClass(P2, [2])
+A divisor class on a normal toric variety
+
+julia> tdc1+tdc2
+A divisor class on a normal toric variety
+```
+"""
+function Base.:+(tdc1::ToricDivisorClass, tdc2::ToricDivisorClass)
+    # check input
+    if !(toric_variety(tdc1) === toric_variety(tdc2))
```

Comment:
Just as a general remark (no need to change this PR), you can also write this:
```suggestion
    if toric_variety(tdc1) !== toric_variety(tdc2)
```